Unable to complete request

**Describe the bug**

when the channel dispatches to some "incorrect" subscriber, the `dispatcher` seems to crash, and logs something like:

```
{"level":"info","ts":"2020-04-30T09:06:17.643Z","logger":"kafkachannel-dispatcher","caller":"kafka/consumer_handler.go:79","msg":"Failure while handling a message","topic":"knative-messaging-kafka.default.kafka-channel-one","partition":0,"offset":82,"error":"unable to complete request to http://channel-display0.default.svc.cluster.local: unexpected HTTP response, expected 2xx, got 400"}
panic: unable to complete request to http://channel-display0.default.svc.cluster.local: unexpected HTTP response, expected 2xx, got 400

goroutine 117 [running]:
knative.dev/eventing-contrib/kafka/channel/pkg/dispatcher.(*KafkaDispatcher).subscribe.func1(0x1ec6d00, 0xc0008a1940, 0xc00073ba50)
	knative.dev/eventing-contrib/kafka/channel/pkg/dispatcher/dispatcher.go:335 +0xb8
created by knative.dev/eventing-contrib/kafka/channel/pkg/dispatcher.(*KafkaDispatcher).subscribe
	knative.dev/eventing-contrib/kafka/channel/pkg/dispatcher/dispatcher.go:333 +0x7a7

```

--- a/kafka/channel/pkg/dispatcher/dispatcher.go
+++ b/kafka/channel/pkg/dispatcher/dispatcher.go
@@ -332,7 +332,7 @@ func (d *KafkaDispatcher) subscribe(channelRef eventingchannels.ChannelReference
 	// this goroutine logs errors incoming
 	go func() {
 		for err = range consumerGroup.Errors() {
-			panic(err)
+			d.logger.Warn("Error in consumer group", zap.Error(err))
 		}
 	}()
 

